c/driver/common/utils.c

```diff
@@ -18,38 +18,178 @@
 #include "utils.h"
 
 #include <errno.h>
-#include <stdarg.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
 #include <assert.h>
+#include "adbc.h"
 
 static size_t kErrorBufferSize = 256;
 
+struct AdbcErrorDetails {
+  char* message;
+
+  char** keys;
+  uint8_t** values;
+  size_t* lengths;
+  int count;
+  int capacity;
+};
+
+static void ReleaseErrorWithDetails(struct AdbcError* error) {
+  struct AdbcErrorDetails* details = (struct AdbcErrorDetails*)error->private_data;
+  free(details->message);
+
+  for (int i = 0; i < 2 * details->count; i++) {
```

Comment:
It seems that we can use `details->count` instead of `2 * details->count` here.

Reply:
ah, good catch - at one point I had encoded both the keys and values in the same array